src/components/MultiselectProblem.tsx

```diff
@@ -1,7 +1,120 @@
-import { BaseProblemProps } from './ProblemSetBlock'
+import { useCallback, useState } from 'react'
+import { mathifyElement } from '../lib/math'
+import { BaseProblemProps, NO_MORE_ATTEMPTS_MESSAGE } from './ProblemSetBlock'
+import { Formik, Field, Form, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
 
-interface MultiSelectProps extends BaseProblemProps {}
+interface MultiselectProps extends BaseProblemProps {
+  solutionOptions: string
+}
+
+interface MultiselectFormValues {
+  response: string[]
+}
+
+export const MultiselectProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback, content, buttonText, solutionOptions,
+  correctResponse, encourageResponse, solution, retryLimit
+}: MultiselectProps): JSX.Element => {
+  const [feedback, setFeedback] = useState('')
+  const [formDisabled, setFormDisabled] = useState(false)
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const solutionArray: string[] = JSON.parse(solution)
+  const parsedOptionValues: string[] = JSON.parse(solutionOptions)
+
+  const schema = Yup.object({
+    response: Yup.array().min(1, 'Please select an answer')
+  })
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+
+  const modifyModel = (values: MultiselectFormValues, e: React.ChangeEvent<HTMLInputElement>): string[] => {
+    const newSet = new Set(values.response)
+    if (e.target.checked) {
+      newSet.add(e.target.value)
+    } else {
+      newSet.delete(e.target.value)
+    }
+    return Array.from(newSet)
+  }
+
+  const generateOptions = (values: MultiselectFormValues, isSubmitting: boolean, setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): JSX.Element[] => {
+    const options: JSX.Element[] = []
+
+    parsedOptionValues.forEach(val => options.push(
+      <>
+    <label className="form-check-label" key={val}>
+      <Field
+      class="form-check-input"
+      type="checkbox"
+      name="response"
+      onChange={(e: React.ChangeEvent<HTMLInputElement>) => { clearFeedback(); setFieldValue('response', modifyModel(values, e)) }}
+      disabled={isSubmitting || formDisabled}
+      value={val}>
+      </Field>
+      {' ' + val}
+    </label>
+    <br></br>
```

Comment:
Same comment as the adding of `' '` above, but I suspect you introduced this for spacing in which case it can be removed. Perhaps you meant to include margins somewhere in which case you should use the bootstrap margin classes.